src/message_interface/message.rs

```diff
@@ -1,28 +1,25 @@
 // Copyright 2022 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-use tokio::sync::mpsc::UnboundedSender;
+use serde::{ser::Serializer, Deserialize, Serialize};
 
-use super::{message_type::MessageType, response::Response};
+use crate::message_interface::ClientMethod;
 
-/// The message type.
-#[derive(Debug, Clone)]
-pub struct Message {
-    pub(crate) message_type: MessageType,
-    pub(crate) response_tx: UnboundedSender<Response>,
+/// The messages that can be sent to the message interface.
+#[derive(Clone, Debug, Deserialize)]
+#[serde(tag = "cmd", content = "payload")]
+pub enum Message {
```

Comment:
Why is this an enum btw ? Do we expect more ?

Reply:
Probably because wallet.rs was first and there we have multiple methods
We could also move all ClientMethods into the Message, but then the serialization part gets really big (actually not sure if it really needs to be done like it's now)